### PR TITLE
feature/esm-apps

### PR DESCRIPTION
**Pull request by jeroenwienk**

### CODE REVIEW

File: lib/App.js [hunk 1]
```diff
@@ -1676,12 +1701,30 @@ $ sudo systemctl restart docker
...
 
+    const templatePath = path.join(__dirname, '..', 'assets', 'templates', 'app', 'drivers');
+
     // Matter devices do not need a device/driver file. As apps cannot add functionality to Matter devices.
     if (!driverWireless.matter) {
-      const fileExtension = App.usesTypeScript({ appPath: this.path }) ? '.ts' : '.js';
-      const templatePath = path.join(__dirname, '..', 'assets', 'templates', 'app', 'drivers');
-      await copyFileAsync(path.join(templatePath, `driver${fileExtension}`), path.join(driverPath, `driver${fileExtension}`));
-      await copyFileAsync(path.join(templatePath, `device${fileExtension}`), path.join(driverPath, `device${fileExtension}`));
+      if (App.usesModules({ appPath: this.path })) {
```
Comment (hmeerlo) [inline lib/App.js:1708]: Sluiten `type: module` en `typescript` elkaar nu uit?

Comment (hmeerlo) [reply lib/App.js:1708]: Of dwingen we af dat typescript altijd `.ts` moet zijn en nooit `.mts` mag zijn?

File: lib/App.js [hunk 2]
```diff
@@ -859,6 +874,9 @@ $ sudo systemctl restart docker
     const hasNodeModules = fs.existsSync(path.join(this.path, 'node_modules'));
     const hasPackageJSON = fs.existsSync(path.join(this.path, 'package.json'));
 
+    fse.ensureDirSync(path.join(this._homeyBuildPath, 'node_modules'));
```
Comment (RobinBol) [inline lib/App.js:877]: Why are these (and below) fs operations sync?

File: lib/App.js [hunk 3]
```diff
@@ -859,6 +874,9 @@ $ sudo systemctl restart docker
...
+
+    // TODO support this?
```
Comment (RobinBol) [inline lib/App.js:879]: What is not supported here exactly?

File: lib/App.js (hunk 2 above)
Comment (RobinBol) [reply lib/App.js:877]: And why must this happen here instead of below with the other fse ensureDirSync calls?

Comment (RobinBol) [reply lib/App.js:877]: > Why are these (and below) fs operations sync?

Ah, I see, this method is executed in parallel.

File: bin/cmds/app/create.js [hunk 4]
```diff
@@ -4,9 +4,16 @@ const Log = require('../../../lib/Log');
 const App = require('../../../lib/App');
 
 exports.desc = 'Create a new Homey App';
+exports.builder = yargs => {
+  return yargs.option('type', {
+    default: 'commonjs',
```
[the remark this answers is not in the document]
Comment (jeroenwienk) [reply bin/cmds/app/create.js:9]: Maybe after a while.

File: lib/App.js (hunk 3 above)
Comment (jeroenwienk) [reply lib/App.js:887]: I ment its odd to have a node_modules folder without a package.json

File: lib/App.js (hunk 1 above)
Comment (jeroenwienk) [reply lib/App.js:1749]: We hebben sowieso nog geen support voor .mts

**Pull request merged**